modules/internal/ChapelDynamicLoading.chpl

```diff
@@ -55,6 +55,7 @@ module ChapelDynamicLoading {
   record chpl_lockWrapper {
     var _lock: chpl_LocalSpinlock;
     inline proc useSpinlock param do return true;
+    inline proc const withLock() ref do return withWriteLock();
     inline proc const withReadLock() ref where useSpinlock do return _lock;
     inline proc const withWriteLock() ref where useSpinlock do return _lock;
```

Comment:
if `useSpinlock` is false this will break.

As a side note, whats the point of having 3 methods that do the exact same thing? Is the intention that in the future they will do different things?

Reply:
For a lock that isn't a RW-lock `withLock()` is always equivalent to `withWriteLock()` since it's meant to take exclusive access.

The idea behind this interface and what i originally used it for is to swap between different locks where `withReadLock()` and `withWriteLock()` do different things.

I may end up doing that again or I may experiment with a different lock interface. But in having this interface, if I decide to change the lock, I don't have to go edit the code.